p2p/peerDenialEvaluator.go

```diff
@@ -0,0 +1,49 @@
+package p2p
+
+import (
+	"time"
+
+	elrondCore "github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go/process"
+)
+
+type peerDenialEvaluator struct {
+	blackListIDsCache          process.PeerBlackListCacher
+	blackListedPublicKeysCache process.TimeCacher
+}
+
+// NewPeerDenialEvaluator creates a new instance of peerDenialEvaluator
+func NewPeerDenialEvaluator(blackListIDsCache process.PeerBlackListCacher, blackListedPublicKeysCache process.TimeCacher) (*peerDenialEvaluator, error) {
```

Comment:
could have ignored blackListedPublicKeysCache as we do not have (yet) a pubkey cacher.
Will remain as it is for now.